### PR TITLE
Justin Maurer's GoNoodle Submission

### PR DESCRIPTION
**Pull request by justinmaurerdotdev**
Thanks for your consideration. Have a great week!

### CODE REVIEW

File: css/src/assets/styles/sass/style.scss [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  -webkit-box-shadow: 0 4px 3px 0 rgba(190,190,190,0.5);
+  box-shadow: 0 4px 3px 0 rgba(190,190,190,0.5);
+  width: 400px;
+  max-width: 100%;
+  margin: 1em auto 0;
+  border-radius: 8px;
+  background: white;
+  cursor: pointer;
+
+  #tile__love {
```
Comment (dcalhoun) [inline css/src/assets/styles/sass/style.scss:32]: I'd recommend avoiding applying of styles to `id`'s. Using `id`'s for styles can cause specificity headaches down the road when working on a large application.

File: css/src/assets/styles/sass/style.scss [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      background: #f5841f;
+      height: 60px;
+      top: 0;
+      left: 0;
+      border-radius: 8px 0 60px 0;
+      border-right: 8px solid white;
+      border-bottom: 8px solid white;
+      padding: 2px;
+    }
+    h2 {
```
Comment (dcalhoun) [inline css/src/assets/styles/sass/style.scss:91]: Styling HTML element directly many times leads to having to override this styles. I'd recommend using a `class` instead.

File: css/src/assets/styles/sass/style.scss [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  }
+  .tile__bottom {
+    position: relative;
+    display: flex;
+    align-items: center;
+    border-top: 1px solid rgba(190,190,190,0.2);
+    padding: 0.75em;
+    color: #999999;
+
+    p.tile__view-more {
```
Comment (dcalhoun) [inline css/src/assets/styles/sass/style.scss:113]: It's generally a good idea to use a little specificity as possible to apply your styles (i.e. – dropping the combination of `p` and `.tile__view-more`). This will allow for more flexibility for your HTML, but also will reduce the likelihood of running into competing levels of specificity.

File: css/src/assets/styles/sass/style.scss [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+          -moz-animation-name: scaleBump;
+          -o-animation-name: scaleBump;
+          animation-name: scaleBump;
+          animation-duration: 0.2s;
+          animation-delay: 0.5s;
+        }
+      }
+    }
+  }
+}
```
Comment (dcalhoun) [inline css/src/assets/styles/sass/style.scss:154]: This bracket, I believe, is closing a Sass selector that is defined 120+ lines above in the file. This likely could be considered merely a personal opinion, but I'd recommend using caution when using Sass' nesting ability. Not only does it increase specificity, but it can sometimes increase the cognitive load when attempting to understand selector you are "within" because the parent selector is likely outside the height of your editor window.

File: javascript/leap.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+var Year = function(yearInput) {
+    this.testYear = yearInput;
+};
+
+Year.prototype.isLeap = function(){
+    var year = this.testYear;
+    if (divisibleBy4(year)) {
+        if (divisibleBy100(year)) {
+            if (divisibleBy400(year)) {
+                return true;
```
Comment (dcalhoun) [inline javascript/leap.js:14]: The several levels nested conditionals might be hard to follow for a coworker. There's likely ways of reducing the nesting, like potentially returning the result of your logical expressions themselves rather than explicit `true`/`false`.

File: javascript/word-count.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+var Word = function() {};
+
+Word.prototype.count = function(phrase){
+	var re = /[\s\n]+/;
+	var phraseArray = phrase.trim().split(re);
+	var wordCounts = {};
+	for ( i=0; i < phraseArray.length; i++ ) {
+		var word = phraseArray[i].toLowerCase();
+		if ( wordCounts.hasOwnProperty(word) ) {
+			for( var key in wordCounts ) {
```
Comment (dcalhoun) [inline javascript/word-count.js:14]: You may be able to get rid of some of this nesting and reduce cognitive load for your fellow developer. Looping through each key in `wordCounts` may be unnecessary if you are checking for the existence of your `word` in `wordCounts`.

**Comment by dcalhoun** [comment]
Thanks for submitting, Justin. I left some inline comments, but here are some high level notes:
- Consistency is important when working on a team. Good job on consistent syntax and style! 👍 
- Some of your CSS contained some relatively high levels of specificity, which can become a big pain in a large code base where you use a given UI component in several different contexts. In my opinion, it's a good practice to use a little specificity as possible to apply your styles.

I'm not very skilled in ruby, so I'll see if I can get a fellow coworker to provide some feedback for you in that area too.

**Comment by justinmaurerdotdev** [comment]
David,
I really appreciate you taking the time to actually review these submission
and provide constructive feedback. I value your suggestions will take all
of them to heart. In terms of the CSS, I definitely understand the
reasoning behind using classes (and other less specific selectors) when
working on a large scale project. I neglected to treat this exercise as a
real-world problem, but I just wanted to be clear that I do understand the
CSS hierarchy and the benefits of using least specific selectors. The JS
recommendations were truly informative and make a lot of sense after
hearing them. And, this was literally my first time working with Ruby, but
I really enjoyed it. It seems a very clean and manageable language upon
first impressions.

Please let me know if there's any way I can help further clarify my skill
set or knowledge level. Have a great week!

## 

Justin Maurer
justin@360zen.com
(512) 585-6364
http://360zen.com

File: ruby/anagram.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+					end
+				end
+				if counter == 0
+					results.push(word)
+				end
+			end
+		end
+		return results.sort
+	end
+end
```
Comment (traviskroberts) [inline ruby/anagram.rb:42]: I think you're making this more complicated than it needs to be. No need to check for `respond_to?` since you're always getting an array. Also, there are some String and Enumerable methods that would vastly simplify your logic (eg `String#chars` and `Enumerable#each_with_object`).

File: ruby/bob.rb [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+	end
+
+	def nothing(prompt)
+		if prompt.strip.empty?
+			return true
+		end
+		return false
+	end
+
+end
```
Comment (traviskroberts) [inline ruby/bob.rb:38]: Overall, this looks really good! The only small change I'd suggest is to make use of Ruby's implicit returns for your test methods (you can get rid of all of your conditionals).
